apps/researcher/src/lib/community.ts

```diff
@@ -0,0 +1,91 @@
+import {clerkClient, auth} from '@clerk/nextjs';
+
+type Community = {
+  id: string;
+  name: string;
+};
+
+export type Membership = {
+  id: string;
+  role: string;
+  publicUserData?: {
+    userId: string;
+    firstName: string | null;
```

Comment:
This is a subset of the Membership model of Clerk. Clerk uses null instead of undefined. Same for the name `publicUserData`. This is the name Clerk uses.

Reply:
I have used the Clerk types instead. So it's clear where these types are coming from.

Reply:
My earlier suggestions were about creating a separation between Clerk and our frontend - as a design principle, to prevent tight coupling. If you think that separation is not necessary, I'm all good 😄